farmfs/snapshot.py

```diff
@@ -1,20 +1,30 @@
-from fs import Path
+from farmfs.fs import Path, rawtype, raw2str
 from func_prototypes import typed
 from delnone import delnone
 from os.path import sep
+from functools import total_ordering
 
+@total_ordering
 class SnapshotItem:
   def __init__(self, path, type, csum=None):
+    if isinstance(type, rawtype):
```

Comment:
doing type coercion here is not good. Lets do this in:

- [ ] reading the tree
- [ ] reading the snaps